.github/workflows/check-and-test.yml

```diff
@@ -0,0 +1,79 @@
+# This action runs lint checks and tests against the code.
+name: Check and Test
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on pushes to master and pull requests for any branch
+  push:
+    branches: [master]
+  pull_request:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  check-and-test:
+    name: Check and Test
+    # The type of runner that the job will run on
+    runs-on: ubuntu-16.04
+
+    strategy:
+      matrix:
+        mongodb_version: [stable, unstable]
+
+    env:
+      MONGODB_VERSION: ${{ matrix.mongodb_version }}
+      MONGODB_TOPOLOGY: standalone
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Adding APT Sources
+        run: sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
```

Comment:
Did not know the `-E` flag 👍 
Regarding repository: When using `ubuntu-20.04` these packages installed later would already be present...

Reply:
I don't think its really needed here, but this is what Travis does and I wanted it to be as close to it as possible (removed it all now anyway 😄)